R/loop_tasks.R

```diff
@@ -88,29 +102,50 @@ loop_tasks <- function(
     # will catch it in the final calls of this function when we run remake on
     # the entire job, properly. if a target is an object rather than a file,
     # we'll build it every time
-    incomplete_targets <- which_incomplete(targets, task_makefile)
     num_targets <- length(incomplete_targets)
+    target_complete <- rep(FALSE, num_targets)
+    
     if(num_targets == 0) break
     
     # if there are remaining targets, try to run them
-    message(sprintf(
-      "\n### Starting loop attempt %s of %s with %s tasks remaining:",
-      this_try, num_tries, num_targets))
+    loop_start_msg <- sprintf(
+      "### Starting loop attempt %s of %s with %s tasks remaining:",
+      this_try, num_tries, num_targets)
+    if (task_msg == 'verbose'){
+      message(paste0("\n", loop_start_msg))
+    } else if (task_msg == 'progress'){
+      pb$message(loop_start_msg)
+    }
     this_try <- this_try + 1
+    
     for(i in seq_len(num_targets)) {
       tryCatch({
         target_num_overall <- incomplete_targets[i]
         target <- targets[target_num_overall]
-        message(sprintf(
-          "Building task %s of %s in loop: %s (#%s of %s total)",
-          i, num_targets, target, target_num_overall, num_targets_overall))
+        if (task_msg == 'verbose'){
+          message(sprintf(
+            "Building task %s of %s in loop: %s (#%s of %s total)",
+            i, num_targets, target, target_num_overall, num_targets_overall))
+        } 
+        
         
         # the main action: run the task-step
         scmake(target, task_makefile, ind_ext=ind_ext, verbose=FALSE)
+        target_complete[i] <- TRUE
         
+        if (task_msg == 'progress'){
+          pb$tick(tokens = list(
+            what = sprintf('%s (#%s of %s)', 
+                           target, target_num_overall, num_targets_overall)))
+        }
       }, error=function(e) {
-        message(sprintf("  Error in %s: %s", deparse(e$call), e$message))
-        message(sprintf("  Skipping task #%s due to error", incomplete_targets[i]))
+        if (task_msg == 'verbose'){
+          message(sprintf("  Error in %s: %s", deparse(e$call), e$message))
+          message(sprintf("  Skipping task #%s due to error", incomplete_targets[i]))
```

Comment:
fine with me if you want to combine these into 1 line. might look funny when the error is long, but probably worth it

Reply:
And then we could have the same message regardless of the value of task_msg, keep the code simple